common/src/com/thoughtworks/go/domain/materials/Modifications.java

```diff
@@ -129,6 +129,11 @@ public boolean shouldBeIgnoredByFilterIn(MaterialConfig materialConfig) {
             return false;
         }
         Set<ModifiedFile> allFiles = getAllFiles(this);
+
+        if (allFiles.isEmpty()) {
```

Comment:
Till now all supported scms' modifications were guaranteed to have modified files. With pluggable scms its not guaranteed. While most plugins should have it, some might not. And in that case auto-schedule pipelines dont run because if modifications dont contain modified files its an ignored modification currently.